File: chapters/04_react_and_tool_stream/v1_direct_answer.py
from __future__ import annotations


def direct_answer(problem: str) -> str:
    if "ModuleNotFoundError" in problem and "requests" in problem:
        return "Install dependency: pip install requests"
    return "Use logs to diagnose the issue."


def main() -> None:
    print(direct_answer("ModuleNotFoundError: No module named 'requests'"))


if __name__ == "__main__":
    main()

File: chapters/04_react_and_tool_stream/v1_direct_answer.ts
// TypeScript mirror for Chapter 04: ReAct and Tool Stream (v1)
// Run with: npx ts-node v1_direct_answer.ts

function directAnswer(problem: string): string {
  if (problem.includes("ModuleNotFoundError") && problem.includes("requests")) {
    return "Install dependency: pip install requests";
  }
  return "Use logs to diagnose the issue.";
}

function main(): void {
  console.log(directAnswer("ModuleNotFoundError: No module named 'requests'"));
}

main();
